app/src/main/java/org/mozilla/reference/browser/library/history/usecases/HistoryUseCases.kt

```diff
@@ -39,6 +39,12 @@ class HistoryUseCases(historyStorage: HistoryStorage) {
         }
     }
 
+    class RemoveFromTopSitesUsesCase(private val historyStorage: HistoryStorage) {
+        operator fun invoke(topSite: TopSite) {
+            (historyStorage as HistoryDatabase).blockDomainsForTopsites(topSite.domain)
```

Comment:
If `historyStorage` is always a `HistoryDatabase` then that should be it's type to make it clear to consumers of this class that that is the only accepted type. You should then do the type cast when creating the class.

Reply:
Do not replace interfaces with real instances, please. This make the use case much harder to test. **Never, never, never replace interfaces with istances** (you can still replace interfaces with extended interfaces.

Reply:
Then there should be an interface with this method in and you should use that instead of casting.